### ISSUE TITLE
Body matcher fails due to regex matching

### ISSUE THREAD

**Issue by nnobelis** [speekha/httpmocker#94]
I have a static scenario with "record". When trying to match them with mode.ENABLED, I get an exception :
```
ERROR fr.speekha.httpmocker.scenario.StaticMockProvider - Scenario file could not be loaded
java.util.regex.PatternSyntaxException: Illegal repetition
{"data":{"username":"XXX","key":"XXX","project_code":"Explorers_Helidon"},"action":"get_information","group":"projects"}
	at java.util.regex.Pattern.error(Pattern.java:1969) ~[?:1.8.0_265]
	at java.util.regex.Pattern.closure(Pattern.java:3171) ~[?:1.8.0_265]
	at java.util.regex.Pattern.sequence(Pattern.java:2148) ~[?:1.8.0_265]
	at java.util.regex.Pattern.expr(Pattern.java:2010) ~[?:1.8.0_265]
	at java.util.regex.Pattern.compile(Pattern.java:1702) ~[?:1.8.0_265]
	at java.util.regex.Pattern.<init>(Pattern.java:1352) ~[?:1.8.0_265]
	at java.util.regex.Pattern.compile(Pattern.java:1028) ~[?:1.8.0_265]
	at kotlin.text.Regex.<init>(Regex.kt:89) ~[kotlin-stdlib-1.4.10.jar:1.4.10-release-411 (1.4.10)]
	at fr.speekha.httpmocker.io.OkHttpExtensionsKt.matchBody(OkHttpExtensions.kt:46) ~[mocker-1.2.0.jar:?]
	at fr.speekha.httpmocker.scenario.RequestMatcher.matchBody(RequestMatcher.kt:37) ~[mocker-1.2.0.jar:?]
```

**To Reproduce**

Create a static scenario with :
```
{
  "request" : {
    "method" : "POST",
    "headers" : { },
    "params" : { },
    "body" : "{\"data\":{\"username\":\"XXX\",\"key\":\"XXX\",\"project_code\":\"Explorers_Helidon\"},\"action\":\"get_information\",\"group\":\"projects\"}"
  },
```

**Expected behavior**
Of course the body section of a request is meant to be a regex, but a recorded scenario should be matched "out of the box" to minimize work for the developer.
**I experimented a little and an easy solution would be to create the regex in OkHttpExtensions.kt:44 :**
`Regex(bodyPattern, RegexOption.LITERAL)`, but then normal regex would not work anymore :(

**Desktop (please complete the following information):**
httpmocker version 1.2.0
Kotlin version 1.4
Retrofit version 2.9.0
OkHttp version 4.8.1

**Comment by speekha** [comment]
Thanks for the feedback. You're right that recorded scenarios should work with as little clean up as possible (although a minimum is probably always needed because ALL request  parameters will be recorded, including some that might be dynamic or constantly changing like authentication tokens for a connected API for instance). As for your suggestion about using RegexOption.LITERAL, could you help me what difference it would make? The doc is minimal on this field, and I'm not sure if it means that the expression will not be considered as an actual regex anymore but more like an exact string to match?
I'm going to look into it as quickly as possible.

**Comment by nnobelis** [comment]
Thanks a lot for the quick response.
As I understand, with RegexOption.LITERAL, the special characters in the string with be ignored and the full string will be used for an exact match : if you have .* in your string, these two characters will be matched.
The problem is that if you always use RegexOption.LITERAL, pattern matching in the body won't be possible any more.

I think the solution is in something like this : if you use RECORD, you get a body like before and it will be matched with RegexOption.LITERAL exactly.
But you can also specify bodyPattern which is indeed a pattern with normal RegEx matching. This bodyPattern takes precedence over body.
Something like that :
```
 "request" : {
    "method" : "POST",
    "headers" : { },
    "params" : { },
    "body" : "{\"data\":{\"username\":\"XXX\",\"key\":\"XXX\",\"project_code\":\"Explorers_Helidon\"},\"action\":\"get_information\",\"group\":\"projects\"}",
    "bodyPattern": ".*get_information.*"
  },

```
But this is a breaking change for your users so you may want to do it "backwards" with bodyRaw for instance and body :)

```
    "bodyRaw" : "{\"data\":{\"username\":\"XXX\",\"key\":\"XXX\",\"project_code\":\"Explorers_Helidon\"},\"action\":\"get_information\",\"group\":\"projects\"}",
    "body": ".*get_information.*"

```

**Comment by speekha** [comment]
That's the way I understand it too: basically, switching the regex to "literal" is like replacing it with a simple String.equals, which is not ideal because it's quite useful to be able to only look for patterns instead of an exact match.
I think I need to look at regex syntax again to find a way to write that body request in a way that won't be mistaken as an incorrect regex syntax. Will keep you posted.

**Comment by sschuberth** [comment]
Can't you just use `Regex.escapeReplacement()` when serializing the `body` to the recorded JSON file, so that any literal characters that have a special meaning in regex are escaped? Then it should work out of the box. However, I'd also renaming the `body` field to `bodyRegex` in the same run to make the expected syntax explicit.

**Comment by speekha** [comment]
That's a good option. That's the kind of solution I'm planning to look into.

**Comment by speekha** [comment]
@nnobelis I've pushed a correction for your issue. I preferred keeping the current API. So request body is converted to a for an exact match regexp. Feel free to test it from develop if you want and make comments before I publish a new version.

**Comment by speekha** [comment]
@nnobelis Correction has been published in 1.3. Let me know if you have any further requests.
